Python/suggestions.c

```diff
@@ -148,6 +148,10 @@ _Py_CalculateSuggestions(PyObject *dir,
     }
     for (Py_ssize_t i = 0; i < dir_size; ++i) {
         PyObject *item = PyList_GET_ITEM(dir, i);
+        if (!PyUnicode_Check(item)) {
+            PyMem_Free(buffer);
```

Comment:
Should you raise an exception here?

Above, `if (dir_size >= MAX_CANDIDATE_ITEMS) { return NULL; }` doesn't raise an exception.

Reply:
I don't think that an exception is needed.